device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/Socks5SocketFactory.java

```diff
@@ -0,0 +1,267 @@
+/*
+ *  Copyright (c) Microsoft. All rights reserved.
+ *  Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ */
+
+package com.microsoft.azure.sdk.iot.device.transport.mqtt;
+
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketAddress;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+
+import javax.net.SocketFactory;
+
+public class Socks5SocketFactory extends SocketFactory
+{
+    private final InetAddress mProxyHost;
+    private final int mProxyPort;
+
+    public Socks5SocketFactory(final String host, final int port) throws UnknownHostException
+    {
+        mProxyHost = InetAddress.getByName(host);
+        mProxyPort = port;
+    }
+
+    @Override
+    public Socket createSocket() {
+        return new Socks5Socket();
+    }
+
+    @Override
+    public Socket createSocket(final InetAddress host, final int port) throws IOException
+    {
+        return createSocketImpl(new InetSocketAddress(host, port));
+    }
+
+    @Override
+    public Socket createSocket(final InetAddress host, final int port, final InetAddress localHost, final int localPort) throws IOException
+    {
+        return createSocketImpl(new InetSocketAddress(host, port), localHost, localPort);
+    }
+
+    @Override
+    public Socket createSocket(final String host, final int port) throws IOException
+    {
+        return createSocketImpl(new InetSocketAddress(host, port));
+    }
+
+    @Override
+    public Socket createSocket(final String host, final int port, final InetAddress localHost, final int localPort) throws IOException
+    {
+        return createSocketImpl(new InetSocketAddress(host, port), localHost, localPort);
+    }
+
+    private Socket createSocketImpl(final InetSocketAddress host, final InetAddress localHost, final int localPort) throws IOException
```

Comment:
It too much overhead. Please remove `createSocketImpl` but call  `createSocket`.

Reply:
Sure thing, I've simplified this a bit